### PR TITLE
Test cases automation for Stable/4.2

### PR DESCRIPTION
**Pull request by nasharatm**
1. Automated test case #OS-2025
2. Reporting related change for barbican test cases which were failing for snapshot mount
3. Added missing test steps for snapshot mount for test#1...4

### CODE REVIEW

**Comment by nasharatm** [comment]
Test case report
image

File: tempest/api/workloadmgr/barbican/test_barbican.py [hunk 1]
```diff
@@ -3582,3 +3586,535 @@ def test_15_barbican(self):
...
         finally:
             reporting.test_case_to_write()
+
+    # Test case automated #OS-2025
+    # http://192.168.15.51/testlink/linkto.php?tprojectPrefix=OS&item=testcase&id=OS-2025
+    @decorators.attr(type='workloadmgr_api')
+    def test_16_barbican(self):
+        try:
+            test_var = "tempest.api.workloadmgr.barbican.with_secret_order_"
+            tests = [[test_var+"create_secret", 0],
```
Comment (priyankagupte15) [inline tempest/api/workloadmgr/barbican/test_barbican.py:3591]: These are steps in test case but here those are reported as different  test cases

Comment (nasharatm) [reply tempest/api/workloadmgr/barbican/test_barbican.py:3591]: We are executing scenario consisting of below tests. Could not attach html report as it is throwing error, hence attached snippet of test execution.

1. tempest.api.workloadmgr.barbican.with_secret_order_create_secret 
2. tempest.api.workloadmgr.barbican.with_secret_order_workload_api
3. tempest.api.workloadmgr.barbican.with_secret_order_full_snapshot_api 
4. tempest.api.workloadmgr.barbican.with_secret_order_incremental_snapshot_api 
5. tempest.api.workloadmgr.barbican.with_secret_order_snapshot_mount_api 
6. tempest.api.workloadmgr.barbican.with_secret_order_filesearch_api
7. tempest.api.workloadmgr.barbican.with_secret_order_selectiverestore_api
8. tempest.api.workloadmgr.barbican.with_secret_order_inplacerestore_api 

tempest.api.workloadmgr.barbican.with_secret_order_oneclickrestore_api

File: tempest/api/workloadmgr/barbican/test_barbican.py [hunk 2]
```diff
@@ -327,25 +330,30 @@ def test_01_barbican(self):
...
                 filesearch_id, "/opt/File_4")
-            for snapshot_id in filecount_in_snapshots.keys():
-                if snapshot_wise_filecount[snapshot_id] == filecount_in_snapshots[snapshot_id]:
-                    filesearch_status = True
-                else:
-                    filesearch_status = False
-
-            if filesearch_status:
-                LOG.debug("Filepath_Search default_parameters successful")
+            if(snapshot_wise_filecount == None):
```
Comment (savitha-p) [inline tempest/api/workloadmgr/barbican/test_barbican.py]: In what case is  None returned by verifyFilepath_Search() ?

File: tempest/api/workloadmgr/barbican/test_barbican.py [hunk 3]
```diff
@@ -488,7 +495,6 @@ def test_01_barbican(self):
                     LOG.debug("***MDSUMS MATCH***")
                     reporting.add_test_step(
                         "Md5 Verification for boot disk", tvaultconf.PASS)
-                    tests[6][1] = 1
```
Comment (savitha-p) [inline tempest/api/workloadmgr/barbican/test_barbican.py:491]: Any specific reason for removing this assignment ?

**Review by priyankagupte15 — APPROVED** [review]

**Comment by savitha-p** [comment]
Also, attach the complete local test run report of all tests.

File: tempest/api/workloadmgr/barbican/test_barbican.py (hunk 3 above)
Comment (nasharatm) [reply tempest/api/workloadmgr/barbican/test_barbican.py:491]: we are updating the var in both "pass and fail" case, it is added at line no.507, hence removed as redundant here.

File: tempest/api/workloadmgr/barbican/test_barbican.py (hunk 2 above)
Comment (nasharatm) [reply tempest/api/workloadmgr/barbican/test_barbican.py]: Removed this code. Not required. It was setup issue and was not needed to be handled.

**Comment by nasharatm** [comment]
Attaching report in pdf format.
[result-sec-order.pdf](https://github.com/trilioData/tempest/files/8618484/result-sec-order.pdf)

**Review by savitha-p — APPROVED** [review]

**Review by SuhasTA4 — COMMENTED** [review]
all changes looks good. 
Approved.

**Pull request merged**